StripeApplePay/StripeApplePay/Source/ApplePayContext/STPApplePayContext.swift

```diff
@@ -145,7 +145,17 @@ public class STPApplePayContext: NSObject, PKPaymentAuthorizationControllerDeleg
     @available(macCatalystApplicationExtension, unavailable)
     @objc(presentApplePayWithCompletion:)
     public func presentApplePay(completion: STPVoidBlock? = nil) {
-        #if STP_BUILD_FOR_VISION
+        // A note on `canImport(CompositorServices)`, found here and elsewhere
+        // in the codebase: This is a terrible, terrible hack.
+        // We should use #if os(visionOS), but that will fail to compile
+        // on Xcode 14. Because we need to continue supporting both Xcode 14
+        // *and* building on visionOS, we instead check `canImport(CompositerServices)`, as CompositerServices currently
```

Comment:
ooc would something like be an alternative?  
```
#if compiler(>=5.9) && os(visionOS) 
```

I kind of did something like this to use an Xcode 15-only type https://github.com/stripe/stripe-ios/blob/2781c735dfd10e7ff17d0c7deba1a98065f85ebb/Stripe/StripeiOS/Source/STPPaymentContext.swift#L334-L335

Reply:
I tried that, but Xcode 14 reads past the `compiler()` directive and fails on `os(visionOS)`. I couldn't figure out a way to structure the second option that didn't involve it being super clunky for `#else` statements.